client/components/swipeable/index.js

```diff
@@ -0,0 +1,212 @@
+import classnames from 'classnames';
+import { Children, useState, useEffect, useRef } from 'react';
+
+import './style.scss';
+
+export const Swipable = ( {
+	hasDynamicHeight = false,
+	children,
+	currentPage = 0,
+	onPageSelect,
+	pageClassName,
+} ) => {
+	const [ isDragging, setIsDragging ] = useState( false );
+
+	const [ pagesStyle, setPagesStyle ] = useState( {
+		transform: `translate3d(0px, 0px, 0px)`,
+		transitionDuration: `300ms`,
+	} );
+	const [ dragStartData, setDragStartData ] = useState( {} );
+
+	const pagesRef = useRef();
+	const resizeObserverRef = useRef();
+	const numPages = Children.count( children );
+
+	const getWidth = () => {
+		return resizeObserverRef.current?.getBoundingClientRect().width;
+	};
```

Comment:
Calling `getBoundingClientRect` can be quite expensive, we should probably look changing this from being a function that is called many times per render, into a memoized value which is recalculated only when necessary

Reply:
I added the calculation and stored the value in the state